front/app/containers/Admin/messaging/routes.tsx

```diff
@@ -13,80 +15,105 @@ const NewSMS = lazy(() => import('./texting/NewSMSCampaign'));
 const PreviewSMS = lazy(() => import('./texting/SMSCampaignPreview'));
 const ExistingSMS = lazy(() => import('./texting/ExistingSMSCampaign'));
 
+enum messagingRoutes {
+  messaging = 'messaging',
+  emailsCustom = `emails/custom`,
+  emailsCustomNew = `emails/custom/new`,
+  emailsCustomCampaignId = 'emails/custom/:campaignId',
+  emailsCustomCampaignIdEdit = 'emails/custom/:campaignId/edit',
+  emailsAutomated = 'emails/automated',
+  texting = 'texting',
+  textingNew = 'texting/new',
+  textingCampaignId = 'texting/:campaignId',
+  textingCampaignIdPreview = 'texting/:campaignId/preview',
+}
+
+export type messagingRouteTypes =
+  | AdminRoute<messagingRoutes.messaging>
+  | AdminRoute<`${messagingRoutes.messaging}/${messagingRoutes.emailsCustom}`>
+  | AdminRoute<`${messagingRoutes.messaging}/${messagingRoutes.emailsCustomNew}`>
+  | AdminRoute<`${messagingRoutes.messaging}/${messagingRoutes.emailsCustomCampaignId}`>
```

Comment:
@EdwinKato I believe here to type the emailsCustomCampaignId we'd need:

AdminRoute<`${messagingRoutes.messaging}/${messagingRoutes.emailsCustom}/${string}`

And then also the same thing (using the` ${string}` )for the other routes with parameters in them?